dvc/repo/fetch.py

```diff
@@ -74,8 +79,8 @@ def fetch(
     except DownloadError as exc:
         failed += exc.amount
 
-    for repo_pair, files in used_external.items():
-        d, f = _fetch_external(self, repo_pair.url, repo_pair.rev, files, jobs)
+    if used_external:
+        d, f = _fetch_external(self, used_external, jobs)
```

Comment:
We will be able to get rid of `used_external` and `_fetch_external()` entirely after push/pull are refactored to use unified `odb.save` (since `save` can handle being passed both regular and external objects)